grind.sh

```diff
@@ -1,132 +1,285 @@
-#! /bin/bash
-
-# this is going to be the SH file to do the work
-############################################################
-# Help                                                     #
-############################################################
-Help()
-{
-   # Display Help
-   echo "Many options to support goatrodeo and ginger processes which are the scan/adg generation and upload respectively."
-   echo
-   echo "Syntax: grind.sh [-b|c|j|k|m|o|p|s|z]"
-   echo ""
-   echo "adg:: Launches goatrodeo generate adgs from container images or applications."
-   echo ""
-   echo "upload:: packages to a Wasabi server using security keys supplied"
-   echo ""
-   echo "options:"
-   echo "-b     For adg, the source directory or file to scan"
-   echo "-c     Command either 'adg' or 'upload'"
-   echo "-j     For upload, the JWT token for uploading to Wasabi"
-   echo "-m     For upload, the mime type of the upload"
-   echo "-o     For adg, directory that output should be directed to"
-   echo "-p     For upload, directory or file that should be uploaded to Wasabi"
-   echo "-z     For upload, the path to a zip file that contains security credentials for Wasabi"
-   echo;
-   exit 1;
+#!/bin/bash
+set -euo pipefail
+
+# Constants
+ADG_MIME_TYPE="application/vnd.cc.bigtent"
+DEPLOYMENT_EVENTS_MIME_TYPE="application/vnd.info.deployevent"
+SPICE_PASS_ENV_VAR="SPICE_PASS"
```

Comment:
I'm not sure that indirection about variable names is helpful. I completely misread how this worked at first and thought you set `SPICE_PASS_ENV_VAR` to the spice pass ... then that `SPICE_PASS_ENV_VAR` was overridable from outside.